src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,113 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private int randomInt;
+
+    private int randomNumberForLegs() {
+        int numericFofLegs;
+        numericFofLegs = random.nextInt(25) + 1;
+        return numericFofLegs;
+    }
+
+    public Figure getDefaultFigure() {
+        Circle circle = new Circle(10);
+        circle.setColor("white");
+        return circle;
+    }
+
+    private int[] isoscelesTrapezoid() {
+        int[] legs = new int[3];
+        int firstLeg = randomNumberForLegs();
+        int secondLeg = randomNumberForLegs();
+        int fourthLeg = randomNumberForLegs() + secondLeg;
+
+        while (firstLeg == secondLeg || secondLeg == fourthLeg
+                || firstLeg + secondLeg < fourthLeg || secondLeg + fourthLeg < firstLeg) {
+            firstLeg = randomNumberForLegs();
+            secondLeg = randomNumberForLegs();
+            fourthLeg = randomNumberForLegs() + secondLeg;
+        }
+
+        legs[0] = firstLeg;
+        legs[1] = secondLeg;
+        legs[2] = fourthLeg;
+        return legs;
+    }
+
+    private int[] rectangle() {
+        int[] legs = new int[2];
+        int firstLeg = randomNumberForLegs();
+        int secondLeg = randomNumberForLegs();
+
+        if (firstLeg != secondLeg && secondLeg > firstLeg) {
+            legs[0] = firstLeg;
+            legs[1] = secondLeg;
+        } else {
+            while (firstLeg != secondLeg && secondLeg > firstLeg) {
+                firstLeg = randomNumberForLegs();
+                secondLeg = randomNumberForLegs();
+            }
+            legs[0] = firstLeg;
+            legs[1] = secondLeg;
+        }
+        return legs;

```

Comment:
explain why my method is worse?

Reply:
Actually, you don't need all these checks to create `Rectangle`
For this task it will be ok to have rectangle with any values for `firstLeg` and `secondLeg`

Reply:
> Actually, you don't need all these checks to create `Rectangle`
> For this task it will be ok to have rectangle with any values for `firstLeg` and `secondLeg`

What if it generates two identical numbers? It's not a rectangle anymore, it's a square.

Reply:
square is a rectangle

Reply:
> square is a rectangle

sorry, where is it written that a square is a rectangle?

Reply:
> A square is a two-dimensional closed shape with 4 equal sides and 4 vertices. Its opposite sides are parallel to each other. We can also think of a square as a rectangle with equal length and breadth.

It is Math rule